adapters/smilewanted/smilewanted.go

```diff
@@ -0,0 +1,162 @@
+				bannerCopy.H = &(firstFormat.H)
+				request.Imp[i].Banner = &bannerCopy
+				validImpExists = true
+			} else if bannerCopy.W != nil && bannerCopy.H != nil {
+				validImpExists = true
+			} else {
+				return nil, []error{&errortypes.BadInput{
+					Message: "No valid banner format in the bid request",
+				}}
+			}
+		} else if request.Imp[i].Video != nil {
+			validImpExists = true
+		}
+	}
+	if !validImpExists {
+		return nil, []error{&errortypes.BadInput{
+			Message: "No valid impression in the bid request",
+		}}
+	}
+
+	request.AT = 1 //Defaulting to first price auction for all prebid requests
+
+	reqJSON, err := json.Marshal(request)
+	if err != nil {
+		return nil, []error{&errortypes.BadInput{
+			Message: fmt.Sprintf("Json not encoded. err: %s", err),
+		}}
+	}
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+	headers.Add("x-openrtb-version", "2.5")
+	headers.Add("sw-integration-type", "prebid_server")
+
+	return []*adapters.RequestData{{
+		Method:  "POST",
+		Uri:     a.URI,
+		Body:    reqJSON,
+		Headers: headers,
+	}}, []error{}
+}
+
+func (a *SmileWantedAdapter) MakeBids(internalRequest *openrtb2.BidRequest, externalRequest *adapters.RequestData, response *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+
+	if response.StatusCode == http.StatusNoContent {
+		return nil, nil
+	}
+
+	if response.StatusCode == http.StatusBadRequest {
+		return nil, []error{&errortypes.BadInput{
+			Message: fmt.Sprintf("Unexpected status code: %d. ", response.StatusCode),
+		}}
+	}
+
+	if response.StatusCode != http.StatusOK {
+		return nil, []error{&errortypes.BadServerResponse{
+			Message: fmt.Sprintf("Unexpected status code: %d. Run with request.debug = 1 for more info", response.StatusCode),
+		}}
+	}
+
+	var bidResp openrtb2.BidResponse
+	if err := json.Unmarshal(response.Body, &bidResp); err != nil {
+		return nil, []error{&errortypes.BadServerResponse{
+			Message: fmt.Sprintf("Bad server response: %d. ", err),
+		}}
+	}
+
+	bidResponse := adapters.NewBidderResponseWithBidsCapacity(len(bidResp.SeatBid[0].Bid))
+	sb := bidResp.SeatBid[0]
+	for i := 0; i < len(sb.Bid); i++ {
+		bid := sb.Bid[i]
+		bidResponse.Bids = append(bidResponse.Bids, &adapters.TypedBid{
+			Bid:     &bid,
+			BidType: getMediaTypeForImp(bid.ImpID, internalRequest.Imp),
+		})
+	}
+	return bidResponse, nil
+}
+
+func addHeaderIfNonEmpty(headers http.Header, headerName string, headerValue string) {
+	if len(headerValue) > 0 {
+		headers.Add(headerName, headerValue)
+	}
+}
+
+// getMediaTypeForImp figures out which media type this bid is for.
+func getMediaTypeForImp(impId string, imps []openrtb2.Imp) openrtb_ext.BidType {
+	mediaType := openrtb_ext.BidTypeBanner //default type
+	for _, imp := range imps {
+		if imp.ID == impId {
+			if imp.Video != nil {
+				mediaType = openrtb_ext.BidTypeVideo
+			}
+			return mediaType
+		}
+	}
+	return mediaType
+}
```

Comment:
This is an anti-pattern. Does your bidding server return the media type of the bid in a bid response extension?

Reply:
I didn't see changes for this comment. The main problem is ensuring the bid type is correct. If your bidding server sees an impression with a banner and a video object, will it always just make a video bid?

Reply:
We always attach a Type on the Zone that we create on our UI and the publisher implement this zone in the ZoneId params, So when we receive a new ad request, we know that it is a request for Display or Video in the zone configuration on our end.

Reply:
Interesting. Would your server return an error if it receives an Imp with both banner and video for a zone that is configured for video?

Reply:
No but we will force this on publisher side to have only one zone per type of request.

We have already done that with PrebidJS and it's works fine ;)

Reply:
Gotcha. Do you think it would be worthwhile to add a check that impressions for your adapter only include banner or video, but not both?

Reply:
Ok so i looked at when I make a config with two types (Banner and Video) and we answer on our side with one ad (the most expensive one between Video and Display on our side), so then PrebidJS displays it ;)

It will be an optimization that we can do later to take into consideration several requests of different types and make several responses, but it requires quite a few changes on our side ;)

Reply:
Since it is not a guaranteed correct identification, one possibility, until you have the infrastructure in place to properly in place to identify bid type, is to set your adapter disabled by default. That will give you a chance to open a dialogue with the PBS host and ensure that the client side has the ability to properly detect and handle the difference between banner and video bids.

Reply:
Just to clarify : We identify on our end the type, we see in each Bid Request BUT we will answer with only one ad (corresponding to the winning ad on our internal competition between all types).

We support multi request but we do not support multi response, but it's only an optimization for us to have more opportunities to win the competition on publisher's side.

Reply:
I think overall the PR LGTM now. Just want to warn you that the above approach of assigning the media type for the bid might lead to the final ad not being rendered properly if the media type wasn't correctly set for a multi-format request. This might happen if for an imp that has both banner and a video object, your server returns a banner bid but then here you assign the media type to be video.

Reply:
We have had issues with other bidders in the past. If the final rendering client is not smart enough, it can get confused as to what it is trying to display. Just considering how you can get ahead of it rather than having your publisher clients have a poor experience that needs to be debugged after the fact.